File: fiberplane-charts/src/Mondrian/generator/utils/normalizeAlongLinearAxis.ts
import type { Axis } from "../../types";

/**
 * Takes an absolute value and normalizes it to a value between 0.0 and 1.0 for
 * the given axis.
 */
export function normalizeAlongLinearAxis(value: number, axis: Axis): number {
  return (value - axis.minValue) / (axis.maxValue - axis.minValue);
}

File: mondrian-charts/src/utils/normalize_along_linear_axis.rs
use crate::types::Axis;

/// Takes an absolute value and normalizes it to a value between 0.0 and 1.0 for
/// the given axis.
pub(crate) fn normalize_along_linear_axis(value: f32, axis: &Axis) -> f32 {
    (value - axis.min_value) / (axis.max_value - axis.min_value)
}
